libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -477,6 +409,56 @@ fn notify_subscribers(
     }
 }
 
+fn send_heartbeats(
+    state: &mut ConnectionState,
+    subscribers: &mut HashMap<String, Sender<Vec<CmNotification>>>,
+) {
+    let heartbeat_message = match create_heartbeat() {
```

Comment:
I'm actually not convinced this can actually ever fail, so this may be a good place to have a panic.

Reply:
made create_heartbeat panic